src/odemis/acq/fastem.py

```diff
@@ -900,6 +915,32 @@ def _create_acquisition_metadata(self):
         self._detector.updateMetadata({model.MD_EXTRA_SETTINGS: json.dumps(selected_settings)})
         return selected_settings
 
+    def _calculate_beam_shift_cor_indices(self, n_beam_shifts=10):
+        """
+        Calculate for which indices to run the beam shift correction. The beam shift correction should run every
+        n sections, starting at the first field in a row. For polygonal sections there can be gaps in the indices,
+        the beam shift correction should then be run on the next possible section.
+        Example: If the beam shift correction should run every 3 sections for the indices [(1, 0), (2, 0), (8, 0)],
+        it should run for indices (1, 0) and (8, 0).
+
+        :param n_beam_shifts: (int) Number of sections after which the beam shift correction should run.
+        """
+        # Sort indices by row first and then by column to enable row-wise processing
+        field_indices = sorted(self._roa.field_indices, key=lambda x: (x[1], x[0]))
+
+        beam_shift_indices = []
+        current_row = -1  # Initialize with an invalid row number to handle the first row properly
+        for idx in field_indices:
+            col, row = idx  # field indices are saved (col, row)
+            if row == current_row + 1:
```

Comment:
Can we just write `if row > current_row`? As the fields are sorted by row, and the whole megafield should be a single shape, I think that should be equivalent, but it feels like clearer to me, and also "safer" in (the impossible?) case the rows would "jump" by more than 1.